client/src/views/Leaderboard.vue

```diff
@@ -86,6 +89,15 @@ export default {
     }
 
     this.isLoading = false
+  },
+  methods: {
+    sortUserLeaderboard(sortingKey) {
+      this.userSortingKey = sortingKey
+      this.userLeaderboard.sort((a, b) => {
+        return b.achievements[sortingKey] - a.achievements[sortingKey]
+      })
+      this.userSortedLeaderboard = this.userLeaderboard.slice(0, 100)
```

Comment:
Recommend to perform the sorting server side, see `/server/services/leaderboard.js`. Might be little tricky to get the sorting right when passing to mongo. I assume we'll want a **default** sort which sorts by rank, then victories, then renown and then a specific sort where the user specifies the sort field and sort direction.

Requesting the entire leaderboard is fine for a small data set but we'll run into issues as the leaderboard gets larger. It also populates the user's guild as well which is another hit.

Reply:
I see, you're right. I originally did it this way to avoid re-requesting the leaderboard from the server, but yes, it's probably better to sort and limit on the server. Will implement this.